backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -49,6 +50,9 @@
    * @return {@link Staff} object which has just been created.
    */
   public Staff createStaff(Staff newStaff) {
+    if (findStaffByEmail(newStaff.getEmail()).isPresent()) {
+      throw new DuplicateEmailException(newStaff.getEmail());
+    }
```

Comment:
Instead of wrapping `isPresent` in an `if`, you can directly use `Optional`'s `ifPresent` function:
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#ifPresent-java.util.function.Consumer-
```
findStaffByEmail(newStaff.getEmail()).ifPresent(
  () -> throw new DuplicateEmailException(newStaff.getEmail())
);
```

I'm also not sure if this logic belongs here or in `StaffRepository.save`. Will we ever want to save a staff if there is duplicate e-mail?

Reply:
This is how I thought I could use `ifPresent`, but I can't seem to get it to work? The code above doesn't compile for me anyway which I find weird. It's exactly how I wrote this initially and couldn't work out why it wasn't working.

> I'm also not sure if this logic belongs here or in StaffRepository.save. Will we ever want to save a staff if there is duplicate e-mail?

Very good point... currently if you update someone you won't get this error, just a JbdcViolationException or something along those lines.

Reply:
Oops, my bad, that's because `ifPresent` takes a `Consumer`, which is a function `Staff -> void`.
The following should work:
```
findStaffByEmail(newStaff.getEmail()).ifPresent(
  (Staff staff) -> {
    throw new DuplicateEmailException(newStaff.getEmail());
  }
);
```

Reply:
Ahhh yeah that's it. I tried putting staff there but it didn't work, forgot that Java sometimes can infer the type. Had this problem before!

Updated anyway, thanks Tad! 👍